### DISCUSSION TITLE
Possibility of adding a DataTable/TableRow api wrapper.

### DISCUSSION THREAD

**Discussion by shortstacked** [getgauge/taiko#2099] [Ideas]
Currently when I'm doing testing a lot of test cases involve validation of data tables, or clicking elements inside of data tables. 

I would like to get some feedback on a potential DataTable and TableRow element wrapper. For instance I can currently access tableCells through  row/column arguments

`tableCell({row:1, column:2))` 

I would like to be able to do other actions such as 
`tableRow({columnName: 'First Name', columnValue: 'shortstacked'})`

Where this would return a full data row, so I can then follow on by accessing things like
`tableRow({columnName: 'First Name', columnValue: 'shortstacked'}).column({name: 'Last Name'}).text()`

This would allow validation of full table rows, or clicking on elements inside of rows, based on other arguments. Would this be worth investigation further?

**Comment by zabil** [comment]
> `tableRow({columnName: 'First Name', columnValue: 'shortstacked'}).column({name: 'Last Name'}).text()`

Under the current design, Taiko's API are flat and should return and element at the first level. In the sample above `tableRow` returns something with an `column` this is not ideal especially to pinpoint which part of the selector failed. 

It's worth experimenting around this as a Taiko table [plugin](https://docs.taiko.dev/writing_plugins/) to figure what a good API for dealing with tables will look like.
